### ISSUE TITLE
v0.9.1: TS2322 Snippet and TS2769 mount() false positives persist

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#105]
svelte-check-rs 0.9.1 still reports 29 false positives that `svelte-check` does not. Two categories:

## 1. TS2322: Snippet type mismatch (11 errors)

Every `{#snippet}` block passed to a component prop typed as `Snippet<[...]>` fails:

```
src/lib/DynamicScatter.svelte:301:5
Error: Type '() => { '{@render ...} must be called with a Snippet': ... }' is not assignable to type 'Snippet<[...]>'. (ts(TS2322))
```

Affected: `DiatomicCurve.svelte`, `DynamicScatter.svelte` (3x), `GitHubActivityScatter.svelte`, `MetricScatter.svelte`, `data/+page.svelte`

## 2. TS2769: No overload matches this call (18 errors)

`mount()` with SvelteKit page components and `HeatmapTable`/`PeriodicTable` components:

```
tests/routes/ContributePage.test.ts:7:17
Error: No overload matches this call. (ts(TS2769))
```

This persists despite the fix in #102 (v0.9.1). The `mount()` errors here are for SvelteKit page components (with `PageData` props), not generic components.

## Reproduction

```bash
git clone https://github.com/janosh/matbench-discovery
cd matbench-discovery/site
pnpm install
pnpm dlx svelte-check-rs    # 29 errors
pnpm dlx svelte-check       # 0 errors (comparison)
```

Branch: `improve-geo-opt-table-and-scatter-plots`

## Environment

- svelte-check-rs: 0.9.1
- svelte: 5.50.0
- typescript: 5.9.3
- @sveltejs/kit: 2.50.2
- matterviz: 0.3.1

**Comment by janosh** [comment]
**Relation to prior issues:**

- **#102** (TS2769 on `mount()`) — was fixed in v0.9.1 for generic components with `HTMLAttributes` intersections. The `mount()` errors here are different: they occur with SvelteKit page components (auto-generated types from `+page.svelte` files) and `HeatmapTable`/`PeriodicTable` from `matterviz` which use complex snippet + bindable prop signatures. The fix in v0.9.1 didn't cover these cases.

- **#94** (Snippet prop typing) — was closed as unable to reproduce with a simple generic snippet component. The Snippet errors here involve `matterviz/plot`'s `ScatterPlot` component which has snippet props typed as `Snippet<[ScatterHandlerProps<Record<string, unknown>>]>` with generic type parameters. The complexity of the upstream component types may be what triggers the false positive.

Both error categories reproduce reliably on the linked branch with a single `pnpm dlx svelte-check-rs` command.

**CI evidence:** These errors are also failing in our PR's `prek` CI check: https://github.com/janosh/matbench-discovery/actions/runs/21962978643/job/63445186151
